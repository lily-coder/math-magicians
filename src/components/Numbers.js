/* eslint no-unused-vars: 0 no-undef: 0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

const Numericals = (props) => {
  const handleClick = (e) => {
    const { handleClick } = props;
    handleClick(e.target.value);
  };

  return (
      <div className='calculator-container numbers-container'>
          <div className='calc-row-five numbers-five'>
            <button value='1' onClick={handleClick}>1</button>
            <button value='2' onClick={handleClick}>2</button>
            <button value='3' onClick={handleClick}>3</button>
        </div>
        <div className='calc-row-four numbers-four'>
            <button value='4' onClick={handleClick}>4</button>
            <button value='5' onClick={handleClick}>5</button>
            <button value='6' onClick={handleClick}>6</button>
        </div>
        <div className='calc-row-three numbers-three'>
            <button value='7' onClick={handleClick}>7</button>
            <button value='8' onClick={handleClick}>8</button>
            <button value='9' onClick={handleClick}>9</button>
        </div>
        <div className='calc-row-six numbers-six'>
            <button className='zero' value='0' onClick={handleClick}>0</button>
            <button className='dot' value='.' onClick={handleClick}>.</button>
        </div>
      </div>
  );
};

Numericals.propTypes = { handleClick: PropTypes.func };

Numericals.defaultProps = { handleClick: null };

export default Numericals;