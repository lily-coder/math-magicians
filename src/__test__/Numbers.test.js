/* eslint-disable no-unused-vars */
import renderer from 'react-test-renderer';
import Numbers from '../components/Numbers.js';

it('renders Numbers Component correctly', () => {
  const tree = renderer.create(<Numbers />).toJSON();
  expect(tree).toMatchSnapshot();
});
