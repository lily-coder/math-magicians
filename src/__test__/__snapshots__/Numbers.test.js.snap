// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Numbers Component correctly 1`] = `
<div
  className="calculator-container numbers-container"
>
  <div
    className="calc-row-five numbers-five"
  >
    <button
      onClick={[Function]}
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      value="3"
    >
      3
    </button>
  </div>
  <div
    className="calc-row-four numbers-four"
  >
    <button
      onClick={[Function]}
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      value="6"
    >
      6
    </button>
  </div>
  <div
    className="calc-row-three numbers-three"
  >
    <button
      onClick={[Function]}
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      value="9"
    >
      9
    </button>
  </div>
  <div
    className="calc-row-six numbers-six"
  >
    <button
      className="zero"
      onClick={[Function]}
      value="0"
    >
      0
    </button>
    <button
      className="dot"
      onClick={[Function]}
      value="."
    >
      .
    </button>
  </div>
</div>
`;
